### PR TITLE
Vfep 603 - add functionality for different types of chapter 33 selections in form 1995

### PR DESCRIPTION
**Pull request by GcioGregg**
## Summary

- vfep-603 - add functionality to show both 1) chapter33 Post911 and 2) chapter33 Fry Scholarship selections in form 1995.  There was an update to vets-json-schema repo to add 2 new enums/values (chapter33post911 and chapter33fryscholarship, replacing the generic "chapter33").  We changed the education_benefits_claim model to account for the new values.
- our team is VFEP
- no flipper or feature flags are used for this change

## Related issue(s)
- [vfep-603](https://vajira.max.gov/browse/VFEP-603), [vfep-597](https://vajira.max.gov/browse/VFEP-597)

## Testing done

- the old behavior was it only showed "chapter33" in the spool file as the benefit selected.  Now the 1995 spool file will show either chapter 33 post 911 or chapter 33 fry scholarship if a chapter 33 benefit is selected.
- to verify changes, submit form 1995 three times, once for each scenario: 1) chapter 33 post911 selected, 2) chapter 33 fryscholarship selected, and 3) any other benefit besides chapter 33 and then review spool files generated to ensure proper selection is displayed as the benefit.
- rspec, rubocop, and developer user tests of form submission and spool files generated.

## What areas of the site does it impact?
form 1995 submission and spool file generation for form 1995

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X]  No error nor warning in the console.
- [X]  Events are being sent to the appropriate logging solution
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/models/education_benefits_claim.rb [hunk 1]
```diff
@@ -118,6 +118,10 @@ def selected_benefits
       benefits['vettec'] = true
     when '1990s'
       benefits['vrrap'] = true
+    when '1995'
+      benefit = parsed_form['benefit']&.underscore
+      benefits['chapter33'] = true if benefit.present? && benefit.start_with?('chapter33')
+      benefits[benefit] = true if benefit.present? && !benefit.start_with?('chapter33')
```
Comment (ryan-mcneil) [inline app/models/education_benefits_claim.rb:124]: Not a problem, but a nice shorthand for these lines could be:
```
      benefits['chapter33'] = true if benefit&.start_with?('chapter33')
      benefits[benefit] = true if !benefit&.start_with?('chapter33')
```

`benefit&.<method>` basically checks the presence of benefit before performing the next method in the chain. Should work for the negation as well!

Comment (nfstern02) [reply app/models/education_benefits_claim.rb:124]: We'll be happy to make that adjustment on a subsequent push, but if you could approve asap as we have a frontend going in that this needs to be in synch with.

**Review by ryan-mcneil — DISMISSED** [review]
Awesome! Thanks for circling back on that!

**Review by ryan-mcneil — APPROVED** [review]
Reapproved 👍

**Pull request merged**